src/pkg/aws/ecs/cfn/template.go

```diff
@@ -283,35 +289,93 @@ func createTemplate(stack, image string, memory float64, vcpu float64, spot bool
 	}
 
 	// 7. ECS task definition
-	cpu, mem := awsecs.FixupFargateConfig(vcpu, memory)
+	var totalCpu, totalMem float64
+	var platform string
+	for _, task := range tasks {
+		totalCpu += task.Cpus
+		totalMem += float64(task.Memory)/1024/1024 + 6 // 6MB for the container overhead
+		if platform == "" {
+			platform = task.Platform
+		} else if platform != task.Platform {
+			panic("all tasks must have the same platform")
+		}
+	}
+	cpu, mem := awsecs.FixupFargateConfig(totalCpu, totalMem)
+	arch, os := awsecs.PlatformToArchOS(platform)
+	var archP, osP *string
+	if arch != "" {
+		archP = ptr.String(arch)
+	}
+	if os != "" {
+		osP = ptr.String(os)
+	}
+
+	var volumes []ecs.TaskDefinition_Volume
+	var containerDefinitions []ecs.TaskDefinition_ContainerDefinition
+	for i, task := range tasks {
+		for _, v := range task.Volumes {
+			volumes = append(volumes, ecs.TaskDefinition_Volume{
+				Name: ptr.String(v.Source),
+			})
+		}
+
+		volumesFrom := make([]ecs.TaskDefinition_VolumeFrom, 0, len(task.VolumesFrom))
+		for _, v := range task.VolumesFrom {
+			parts := strings.SplitN(v, ":", 2)
+			ro := false
+			if len(parts) == 2 && parts[1] == "ro" {
+				ro = true
+			}
+			volumesFrom = append(volumesFrom, ecs.TaskDefinition_VolumeFrom{
+				ReadOnly:        ptr.Bool(ro),
+				SourceContainer: ptr.String(parts[0]),
+			})
+		}
+
+		mountPoints := make([]ecs.TaskDefinition_MountPoint, 0, len(task.Volumes))
+		for _, v := range task.Volumes {
+			mountPoints = append(mountPoints, ecs.TaskDefinition_MountPoint{
+				ContainerPath: ptr.String(v.Target),
+				SourceVolume:  ptr.String(v.Source),
+				ReadOnly:      ptr.Bool(v.ReadOnly),
+			})
+		}
+
+		def := ecs.TaskDefinition_ContainerDefinition{
+			Name:        task.Name,
+			Image:       images[i],
+			StopTimeout: ptr.Int(120),
+			Essential:   task.Essential,
+			LogConfiguration: &ecs.TaskDefinition_LogConfiguration{
+				LogDriver: "awslogs",
+				Options: map[string]string{
+					"awslogs-group":         cloudformation.Ref(_logGroup),
+					"awslogs-region":        cloudformation.Ref("AWS::Region"),
+					"awslogs-stream-prefix": awsecs.AwsLogsStreamPrefix,
+				},
+			},
+			VolumesFrom:      volumesFrom,
+			MountPoints:      mountPoints,
+			EntryPoint:       task.EntryPoint,
+			Command:          task.Command,
+			WorkingDirectory: ptr.String("/app"), // TODO: make this configurable
```

Comment:
add to `Task`?

Reply:
Added both WorkDir and DependsOn to the Container (previously task) struct